### DISCUSSION TITLE
Temporary avoid i.e. toll roads and/or motorways (country wise)?

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#607] [Q&A] [answered]
Hi, I just came back from a tour where I used Cruiser for more then 6tkm. In General it worked great. Since I crossed several countries with several motorways and/or toll regulations, I missed the function to turn i.e. the usage from motorways or toll roads temporary off.
I know, I can set this functions in BRouter but then it's valid for the whole tour/trip.
It would be nice to have this function available like "skip next waypoint" so just for turning on/of on demand best (but probably more complicated is,) having this option additional country wise.
Cheers, Dirk

**Comment by devemux86** [comment]
@DIrgi thanks for the suggestion.

Do you mean in route planning or navigation?

In planning the routing options can be set per route profile (fastest, curvy, etc)
and each route section (between 2 waypoints) can have different route profiles.

**Reply by DIrgi** [reply]
In general during navigation to force Cruiser on demand to do a recalculation to avoid toll roads and/or motorways.
Yes I know, that for planning or in general I can set i.e. don't use motorways and so on.

Reason for request behind.
I used to drive a couple of days ago in Austria and Germany. In Austria you have to pay toll for motorways even as a biker.
Additional there are some special toll roads everybody has to pay i.e. for a mountain road.

- When I say in the settings: Don't use motorways to (avoid toll motorways). It has as well an influence on German motorways where my trip continues after border crossing.
- If I select in the settings, don't use toll roads, I get blocked on Austrian motorways but not on German ones since there's no toll (so far so good) but, I got blocked for every toll road even mountain roads.

------------------------
When planning the tour on the Cruiser App, I didn't set any avoidance but...tried to do by using shaping points to bypass toll motorways but everywhere I did not look closer, cruiser guided me on the motorways. So this was annoying but I don't want to switch block toll roads in general.

I remember ones in my VW car navigation I could select the option by country like: avoid motorways (only) in Austria.

However it would be nice to have the option while navigation, to fast switch country wise don't use motor ways, and or don't use toll roads.

devemux86: _"...and each route section (between 2 waypoints) can have different route profiles..."_
But this needs lot of fine tuning and time. Better would be country wise and just by fast click.

Cheers, Dirk

**Reply by devemux86** [reply]
> it would be nice to have the option while navigation

@DIrgi What do mean in navigation?

Navigation only does **re-routing** from user location to next waypoint with active routing options.
This happens when off route (automatically or manually), skip waypoint, avoid road block, etc.

**Reply by DIrgi** [reply]
Well, while I follow the route I have previously planned by setting waypoints (Cruiser navigates me = navigation mode) this I mean with naviation. So while I'm driving and following the route I would like to have the option to press a button (like delete next waypoint) to avoid i.e. toll roads as long as I deactivate it (and best in addition with country like: avoid toll roads in Austria)
If I press the button Cruiser navigations is/should be forced the to avoid any motorways and do a recalculation if necessary.

**Reply by devemux86** [reply]
@DIrgi 

Country specific routing rules are difficult to be done automatically.
Even the side of the road is selected manually by users, for routers that need such separation.

So you mean a _rerouting until next waypoint_ but with different options than the currently active?

**Reply by DIrgi** [reply]
> @DIrgi

> So you mean a _rerouting until next waypoint_ but with different options than the currently active?

No, not really, I mean e re-routing (at least as a country wise selection will not work then) until I say I will stop and not just to the next waypoint.

The effect will be like I start a new routing but now with new preliminaries

**Reply by devemux86** [reply]
@DIrgi  Do you mean route planning? Plan a different route and restart navigation.

Or do you mean something else?
Re-routing requires start + end and merge with the existing route.

So to see for any possible changes, we need to understand with details the steps?

**Reply by DIrgi** [reply]
somehow I'm despairing right now...is it really so difficult to understand?
Ok, I try do show it with screen shoots.

you see "a part" of a route. Since there are no restrictions set in the options,
Cruiser is guiding me via the motorway which is the fastest.

![image](https://github.com/user-attachments/assets/070d736b-4a86-43ee-a2be-effbfc1b3987)


When I'm following the route and recognize that Cruiser will start to join the motorway although I don't want it at the moment,
I press that "avoid button" and cruiser (should) start recalculating and guide me via the next possible country road.

![image](https://github.com/user-attachments/assets/0a080d73-d896-4da3-99bf-ddf20dae8afa)

**Reply by devemux86** [reply]
@DIrgi thanks for the detailed description, we need the exact steps to understand the reports.

> When I'm following the route

If you mean navigation, then it is **rerouting** that can only be done from user location to a next waypoint.

So you are asking for a button to calculate a different route until a next waypoint.
With what options, "Avoid motorways" + "Avoid toll roads"?

Note that the result varies, e.g. if you are already on motorway it could try to find the next exit, etc.

Similar to route planning, when change the active routing options and select "Recalculate section".

**Reply by DIrgi** [reply]
Sorry, I was a little bit absence the last days.
It should not only be valid until the next waypoint it should be valid as long as I temporary activate it. Best would be country wise if not it's as well okay without.

It should just be just a quick support while I'm driving and recognize that my routing (I haven't checked that detailed before) will guide me on a motorway which I don't want or as well on a toll road. I know in the settings I can switch but then it's valid for all calculations what I don't want in general. Normally I use my waypoints to force routing to go where I want.

So just something like in google maps. When I select avoid motorway or toll road it does immediately e recalculation of an existing tour as long as avoid is selected.

**Reply by devemux86** [reply, marked as answer]
> When I select avoid motorway or toll road it does immediately e recalculation of an existing tour as long as avoid is selected.

Inside navigation this is rerouting and is done until the next waypoint or a selected next waypoint.

To avoid the next km / mi of a specific road or tolls can also use the "Avoid road block" fox X km / mi.

**Comment by MotoUKRider** [comment]
A quick test on "Toll Roads" shows that there is still an [outstanding BRouter issue](https://github.com/abrensch/brouter/issues/639) regarding "Motorcycle" vehicle types when the road allows them to pass "toll-free", ie: only cars have to pay.

**Reply by devemux86** [reply]
@MotoUKRider 

BRouter needs to update its routing data, I asked here: https://github.com/abrensch/brouter/pull/458#issuecomment-2260083649

**Reply by DIrgi** [reply]
I didn't think so far - yes indeed there are countries you don't pay as a motorbiker but you pay as car driver.
for example: Chechia, Norway, Romania, Slowakia.......

**Reply by MotoUKRider** [reply]
@DIrgi It is worth checking the bridge/road properties on OpenStreetMap as that's how I found this "bug" when trying to work out why I was being kept away from a particular bridge. I had to set the motorcycle-toll property to "no" and wait a day for the BRouter update - then I tried it: Nout.  Feedback from @devemux86 said the code didn't support the vehicle type during lookup.

So, the worst case scenario is, using BRouter, sending you on an unnecessary detour to avoid a bridge/road that it *thinks* you want to avoid paying for but could actually pass for free.

This goes back to last October.
